ilastik/workflows/carving/opCarving.py

```diff
@@ -348,20 +335,21 @@ def restore_and_get_labels_for_object(self, name):
         self._mst.setResulFgObj(fgNodes[0])
 
         self._setCurrObjectName(name)
-        self.HasSegmentation.setValue(True)
+        self._updateCanObjectBeSaved()
 
         # now that 'name' is no longer part of the set of finished objects, rebuild the done overlay
         self._buildDone()
         return (fgVoxelsSeedPos, bgVoxelsSeedPos)
 
     def loadObject(self, name):
-        logger.info("want to load object with name = %s" % name)
-        if not self.hasObjectWithName(name):
-            logger.info("  --> no such object '%s'" % name)
-            return False
+        logger.info(f"want to load object with name = {name}")
+        if name not in self._mst.object_lut:
+            logger.info("  --> no object with this name")
+            return
 
-        if self.hasCurrentObject():
-            self.saveCurrentObject()
+        if self.CanObjectBeSaved.value:
```

Comment:
Should we do this silently without giving the user to intervene? This will save an object with the default name...

Reply:
I wonder if it's possible to make `<not saved yet>` purely a view thing, and deal with an "empty" name (`""` or `None`) in all other logic code?..

Reply:
I agree; I've pushed an update to get a user confirmation to discard current (potentially unsaved) work instead.

Also agree that I'd prefer the default string to be strictly GUI-side